src/arkode/arkode_mristep.c

```diff
@@ -1462,6 +1465,30 @@ int mriStep_TakeStep(void* arkode_mem, realtype *dsmPtr, int *nflagPtr)
          ark_mem->nst, ark_mem->h, ark_mem->tcur);
 #endif
 
+  /* if fullrhs has not been called, fill in Fse[0] and Fsi[0] as applicable */
+  if (!ark_mem->call_fullrhs_start) {
```

Comment:
What is the advantage of your proposal?

Reply:
I think this would make the code more consistent. `call_fullrhs_start` is really more like call the full RHS at initialization and then this call (like in ERKStep) never actually happens and the full RHS is actually updated at the end of the step.